pennylane_lightning/src/util/Kokkos_Sparse.hpp

```diff
@@ -0,0 +1,198 @@
+ * @param vector_size   size of the vector.
+ * @param row_map_ptr   pointer to the row_map.
+ *                      The j element encodes the number of non-zeros above row
+ * j.
+ * @param row_map_size  number of elements in the row_map.
+ * @param entries_ptr   pointer to the column indices of the non-zero elements.
+ * @param values_ptr    non-zero elements.
+ * @param numNNZ        number of non-zero elements.
+ * @return std::vector<data_type<fp_precision>>
+ */
+template <class fp_precision, class index_type>
+void apply_Sparse_Matrix_Kokkos(
+    const std::complex<fp_precision> *vector_ptr, const index_type vector_size,
+    const index_type *row_map_ptr, const index_type row_map_size,
+    const index_type *entries_ptr, const std::complex<fp_precision> *values_ptr,
+    const index_type numNNZ, std::vector<std::complex<fp_precision>> &result) {
+
+    Kokkos::initialize();
+    {
+        const_data_view_type<fp_precision> vector_view(
+            reinterpret_cast<const Kokkos::complex<fp_precision> *>(vector_ptr),
+            vector_size);
+        Kokkos::View<data_type<fp_precision> *> result_view("rhs", vector_size);
+
+        const_crs_matrix_type<fp_precision> myMatrix =
+            create_Kokkos_Sparse_Matrix(row_map_ptr, row_map_size - 1,
+                                        entries_ptr, values_ptr, numNNZ);
+
+        const data_type<fp_precision> alpha(1.0);
+        const data_type<fp_precision> beta;
+        KokkosSparse::spmv("N", alpha, myMatrix, vector_view, beta,
+                           result_view);
+
+        result = std::move(std::vector<std::complex<fp_precision>>(
+            reinterpret_cast<std::complex<fp_precision> *>(result_view.data()),
+            reinterpret_cast<std::complex<fp_precision> *>(result_view.data()) +
+                result_view.size()));
+    }
+    Kokkos::finalize();
+};
+
+} // namespace Pennylane::Util
+#else
+constexpr bool USE_KOKKOS = false;
+namespace Pennylane::Util {
+/**
+ * @brief Apply a sparse matrix to a vector with Kokkos.
+ *  Abort, because there is no Kokkos and Kokkos Kernels installation.
+ *
+ * @tparam fp_precision
+ * @return std::vector<data_type<fp_precision>>
+ */
+template <class fp_precision, class index_type>
+void apply_Sparse_Matrix_Kokkos(
+    [[maybe_unused]] const std::complex<fp_precision> *vector_ptr,
+    [[maybe_unused]] const index_type vector_size,
+    [[maybe_unused]] const index_type *row_map_ptr,
+    [[maybe_unused]] const index_type row_map_size,
+    [[maybe_unused]] const index_type *entries_ptr,
+    [[maybe_unused]] const std::complex<fp_precision> *values_ptr,
+    [[maybe_unused]] const index_type numNNZ,
+    [[maybe_unused]] std::vector<std::complex<fp_precision>> &result,
+    bool throw_exception = true) {
+    if (throw_exception) {
+        PL_ABORT("Executing the product of a Sparse matrix and a vector needs "
+                 "Kokkos and Kokkos Kernels installation.");
+    }
+};
```

Comment:
how does this function calculate the product? I maybe missing something, but it seems that it doesn't do anything.